### ISSUE TITLE
"lxc profile delete foo" succeeds when 'foo' doesn't exist

### ISSUE THREAD

**Issue by jameinel** [lxc/lxd#1886]
I'm not sure if it was intentionally designed this way, but I accidentally had a typo in my command, and then it came back and told me that it successfully deleted the profile. Personally, I feel like that gives people a false sense. Even if the command still exited with Success the wording should be different.

This is with 2.0.0rc9:
$ lxc profile delete blablabla
Profile blablabla deleted
$ lxc profile delete blablabla
Profile blablabla deleted

Even if it *did* exist the first time, it certainly couldn't have been deleted the second time.

**Comment by srkunze** [comment]
Why not?

**Comment by stgraber** [comment]
We should aim for consistency here and the other commands do fail when you're trying to delete or unset something which doesn't exist.

I believe in another report, I did say that I would be happy with us using say "2" as the return code for "doesn't exist" so that shell scripts can run the command without checking first and only fail if they get "1" as the return code (which would indicate an actual failure).

**Comment by srkunze** [comment]
So, this means, you would also accept Success (aka 0) if all functions would do it that way? ;)

Just my 2 cents here: I think it's not 100% compatible with the HTTP verbs/resource handling which LXD tries to get right. Maybe, I get this wrong for the console, because there we should be cautious. I am not sure.

**Comment by jameinel** [comment]
From a user's perspective, if I try to delete something but enter a typo
and the command tells me "successfully deleted" then I believe I did what I
wanted.  But I didn't because the real profile still exists. If it told me
"no such profile" then I can determine whether I already deleted it, or
notice that I entered the wrong name.
I'm not as concerned about the error code for this use case as that is more
for scripts than for a human.

John
=:->
On Apr 12, 2016 20:46, "Sven R. Kunze" <notifications@github.com> wrote:

So, this means, you would also accept Success (aka 0) if all functions
would do it that way? ;)

Just my 2 cents here: I think it's not 100% compatible with the HTTP
verbs/resource handling which LXD tries to get right. Maybe, I get this
wrong for the console, because there we should be cautious. I am not sure.

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub
<https://github.com/lxc/lxd/issues/1886#issuecomment-209002005>
